src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -387,8 +394,13 @@ export class StudyViewPageStore {
     }
 
     @action
-    updateCNAGeneFilters(filters: CopyNumberGeneFilterElement[]) {
-        this._cnaGeneFilter = _.xor(this._cnaGeneFilter, filters);
+    addCNAGeneFilters(filters: CopyNumberGeneFilterElement[]) {
+        this._cnaGeneFilter = _.reduce(this._cnaGeneFilter, function (acc, next) {
```

Comment:
even here `this._cnaGeneFilter = [...this._cnaGeneFilter, {alterations: filters}]`